tests/unit/test_schema_registry.py

```diff
@@ -0,0 +1,257 @@
+
+import logging
+
+LOG = logging.getLogger(__name__)
+
+
+class MockedSchemaRegistry:
+    def __init__(self, compatibility_mode: CompatibilityModes, has_deleted_schema: bool = False):
+        self.schema_registry = KarapaceSchemaRegistry(DEFAULTS)
+        self.compatibility_mode = compatibility_mode
+        self.schema_registry.get_compatibility_mode = Mock(return_value=self.compatibility_mode)
+
+        schema_version1 = Mock(spec=SchemaVersion)
+        schema_version1.deleted = False
+        typed_schema1 = Mock(spec=TypedSchema)
+        schema_version1.schema = typed_schema1
+
+        schema_version2 = Mock(spec=SchemaVersion)
+        schema_version2.deleted = has_deleted_schema
+        typed_schema2 = Mock(spec=TypedSchema)
+        schema_version2.schema = typed_schema2
+
+        schema_version3 = Mock(spec=SchemaVersion)
+        typed_schema3 = Mock(spec=TypedSchema)
+        schema_version3.deleted = False
+        schema_version3.schema = typed_schema3
+
+        self.schema_registry.database = Mock(spec=InMemoryDatabase)
+
+        self.schema_registry.database.find_subject_schemas = Mock(
+            return_value={
+                Version(1): schema_version1,
+                Version(2): schema_version2,
+                Version(3): schema_version3,
+            }
+        )
+
+        self.parsed_schema1 = Mock(spec=ParsedTypedSchema)
+        self.parsed_schema2 = Mock(spec=ParsedTypedSchema)
+        self.parsed_schema3 = Mock(spec=ParsedTypedSchema)
+
+        def resolve_and_parse_mock(schema: TypedSchema) -> ParsedTypedSchema:
+            if schema == typed_schema1:
+                return self.parsed_schema1
+            if schema == typed_schema2:
+                return self.parsed_schema2
+            if schema == typed_schema3:
+                return self.parsed_schema3
+            raise ValueError(f"Unexpected object {schema}")
+
+        self.schema_registry.resolve_and_parse = MagicMock(side_effect=resolve_and_parse_mock)
+
+    def check_schema_compatibility(
+        self,
+        new_schema: ValidatedTypedSchema,
+        subject: Subject,
+    ) -> SchemaCompatibilityResult:
+        return self.schema_registry.check_schema_compatibility(new_schema, subject)
+
+
+async def test_schema_compatible_in_transitive_mode() -> None:
+    # Given
+    schema_registry = MockedSchemaRegistry(CompatibilityModes.FULL_TRANSITIVE)
+    new_schema = ValidatedTypedSchema.parse(SchemaType.JSONSCHEMA, '{"type": "array"}')
+    subject = Subject("subject")
+
+    # Don't test the actual compatibility checks
+    result = Mock(spec=SchemaCompatibilityResult)
+    result.compatibility = SchemaCompatibilityType.compatible
+    SchemaCompatibility.check_compatibility = Mock(return_value=result)
+
+    # When
+    schema_registry.check_schema_compatibility(new_schema, subject)
+
+    # Then
+    assert result.compatibility is SchemaCompatibilityType.compatible
+
+    # All 3 schemas are checked against
```

Comment:
got it, but why not creating a test where you know that the schemas are compatible 1 by 1 but not fully transitive? Isn't better to test the outcome instead of the actions? Again, matter of taste. Maybe better to ask a second opinion on this before doing work :D Maybe its me that isn't used to test like this and isn't anything bad

Reply:
Moved to `tests/integration/test_schema_compatibility.py`